.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -789,6 +789,18 @@ private DateTimeResolutionResult ParseDuration(string text, DateObject reference
                         beginDate = DurationParsingUtil.ShiftDateTime(durationResult.Timex, endDate, false);
                     }
 
+                    // Handle the "within two weeks" case which means from today to the end of next two weeks
+                    prefixMatch = config.WithinConnectorRegex.Match(beforeStr);
+                    if (prefixMatch.Success && prefixMatch.Length == beforeStr.Length &&
+                        DurationParsingUtil.IsDateDuration(durationResult.Timex))
+                    {
+                        GetModAndDate(ref beginDate, ref endDate, referenceDate, durationResult.Timex, true, out mod);
+
+                        // In GetModAndDate, if this is the "future" case, the beginDate/endDate add one day; For the "within" case, after discussion, we don't need to add this one day.
```

Comment:
I don't understand this comment.

Reply:
This means for other future time cases, the beginDate would add one day automatically (such as the "in" case).
But as discussed, the "within" case should start today.

Reply:
Please make the 'true' a named parameter. And change the code comment to something like:
// In GetModAndDate, this "future" resolution will add one day to beginDate/endDate, but for the "within" case it should start from the current day.

Reply:
Fixed